### PR TITLE
drivers: npm6001: create mfd device to handle common init code

### PR DESCRIPTION
**Pull request by gmarull**
This PR adds the skeleton for MFD device drivers and moves common init code for nPM6001 from regulator driver to a new mfd driver. Because nPM6001 drivers just use plain I2C API, the MFD driver is API-less. Other MFD devices may provide common register access calls.

### CODE REVIEW

**Comment by aasinclair** [comment]
Is the intention for this pattern to be used for other regulators going forward?

**Review by fabiobaltieri — APPROVED** [review]

File: drivers/gpio/Kconfig.npm6001 [hunk 1]
```diff
@@ -6,13 +6,13 @@ config GPIO_NPM6001
...
 
 config GPIO_NPM6001_INIT_PRIORITY
 	int "nPM6001 GPIO driver initialization priority"
 	depends on GPIO_NPM6001
-	default 60
+	default 65
 	help
-	  Initialization priority for the nPM6001 GPIO driver. It must be
-	  greater than the I2C controller init priority.
+	  Initialization priority for the nPM6001 GPIO driver.
```
[the remark this answers is not in the document]
Comment (fabiobaltieri) [reply drivers/gpio/Kconfig.npm6001:18]: Actually more of a question, are the actual class device nodes going to be a child of `nordic,npm6001`? I think that's how Linux does it.

**Review by aasinclair — APPROVED** [review]

File: drivers/gpio/Kconfig.npm6001 (hunk 1 above)
[the remark this answers is not in the document]
Comment (gmarull) [reply drivers/gpio/Kconfig.npm6001:18]: This is the current DT representation (modeled following other PMICs in Linux):

https://github.com/zephyrproject-rtos/zephyr/blob/defb159ab1c45eb58ca12dcf898b2cbbed6b2ec6/boards/shields/npm6001_ek/npm6001_ek.overlay#L7-L55

So the day we use DT ordinals and not manual priorities everything should work as expected.

[the remark this answers is not in the document]
Comment (fabiobaltieri) [reply drivers/gpio/Kconfig.npm6001:18]: Cool, yeah then static checking should work as well, good stuff.

**Pull request merged**